hi-ml-cpath/src/health_cpath/preprocessing/loading.py

```diff
@@ -71,9 +72,27 @@ def __init__(
         self.level = level
         self.margin = margin
 
-    def _get_bounding_box(self, slide_obj: Any) -> box_utils.Box:
+    def _get_foreground_mask(self, slide_obj: Any, highest_level: int) -> np.ndarray:
+        """Estimate foreground mask at the highest resolution (i.e. lowest level) of the slide."""
         raise NotImplementedError
 
+    def _get_whole_slide_bbox(self, slide_obj: Any, highest_level: int) -> box_utils.Box:
+        """Return a bounding box that covers the whole slide at the highest resolution (i.e. lowest level)."""
+        h, w = self.reader.get_size(slide_obj, level=highest_level)
+        return box_utils.Box(0, 0, w, h)
+
+    def _get_bounding_box(self, slide_obj: Any, slide_id: int) -> box_utils.Box:
+        """Estimate bounding box at the lowest resolution (i.e. highest level) of the slide."""
+        highest_level = self.reader.get_level_count(slide_obj) - 1
+        scale = self.reader.get_downsample_ratio(slide_obj, highest_level)
+        foreground_mask = self._get_foreground_mask(slide_obj, highest_level)
+        try:
+            bbox = box_utils.get_bounding_box(foreground_mask)
+        except RuntimeError as e:
+            print_message_with_rank_pid(f"Failed to estimate bounding box for slide {slide_id}: {e}")
```

Comment:
This is useful to find out offending slides. We had quite a few jobs failing because of this error and couldn't find out which slide until we rerun the experiment (hard to debug: sometimes it happens after 4h of training)